src/Common/Request/Abstract_Query_Var.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * Abstract class for query vars.
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Request
+ */
+
+namespace TEC\Common\Request;
+
+use TEC\Common\Contracts\Provider\Controller;
+
+/**
+ * Abstract class for query vars.
+ * Contains functions for filtering and sanitizing query vars.
+ *
+ * Extend this class to create a new query var object.
+ *
+ * @see Query_Vars::register()
+ * @see Query_Vars::unregister()
+ *
+ * @since TBD
+ *
+ * @package TEC\Common\Request
+ */
+abstract class Abstract_Query_Var extends Controller {
+	/**
+	 * The query var name (key/slug).
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	protected string $name;
+
+	/**
+	 * Whether the query var should be filtered.
+	 *
+	 * @since TBD
+	 *
+	 * @var bool
+	 */
+	protected bool $should_filter = false;
+
+	/**
+	 * Registers the query var filtering.
+	 *
+	 * @since TBD
+	 *
+	 * @return void
+	 */
+	protected function do_register(): void {
+		$this->container->singleton( $this->name, $this, [ 'register' ] );
```

Comment:
What are you trying to do here ?

$this is already initialized and the register method has already been called. You just want to bind $this->name to the same implementation $this ?